cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -338,6 +338,16 @@ culpa qui officia deserunt mollit anim id est laborum.
   }
 };
 
+// A class that exists only to expose
+// AzureFileSystem::ForceCachedHierarchicalNamespaceSupport to unit tests.
```

Comment:
Or you can use a `friend TestAzureFileSystem` declaration in `AzureFileSystem` for less hackery.

Reply:
I tried and it didn't work. But now I realize it might have been because `TestAzureFileSystem` is inside an anonymous namespace. :thinking:

Reply:
Yep. The sneaky `namespace {` that existed in `azurefs_test.cc` and I didn't notice yesterday when trying to make these classes friends with each other.